### ISSUE TITLE
range-for

### ISSUE THREAD

**Issue by NotAPenguin0** [NotAPenguin0/pscript#4]
Basic range-for syntax:

```rust
let my_list = [1, 2, 3];
for (let val : my_list) {
   std.io.print(val);
}
```

Ideally we also expose a library function `std.iter.range()`, so we can write

```rust
for (let i : std.iter.range(100) {
   // i from 0 to 100
}
```

This could be done by making `range()` simply return a list but this seems quite slow and inefficient. An `iterable` type would perhaps be more useful and flexible. Implementing this in a flexible way isn't trivial though.

**Comment by NotAPenguin0** [comment]
range syntax was decided on to be 
```rust
for (let i : 0..100) {

}
```
instead of having an iterable type.

**Comment by NotAPenguin0** [comment]
Implemented in [23a58f8](https://github.com/NotAPenguin0/pscript/commit/23a58f88d949144da000fd94052c8361ba84d582)
